app/src/main/java/edu/uc/groupProject/topten/ui/main/MainViewModel.kt

```diff
@@ -20,50 +23,153 @@ import edu.uc.groupProject.topten.Service.StrawpollService
+            ListItem("The Godfather", "n/a", 24),
+            ListItem("The Avengers", "They all team up to fight bad guys", 231),
+            ListItem("Inception", "", 12),
+            ListItem("E.T", "", 124),
+            ListItem("The Matrix", "", 42)
+        )
+        return sampleListItems
     }
-  
+
     /**
-     * This function fires every time a vote is added or changed on the generated list. Responsible
-     * for creating DTO objects from the database list.
+     * Provide the user name from the firebase database
+     * @return userName
      */
-    private var firestore : FirebaseFirestore = FirebaseFirestore.getInstance()
-    init{
-        firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
-        waitForListUpdate()
+    fun getUserName(): String {
+        val docRef = firestore.collection("users").document("testuser")
+        var userName: String = ""
+
+        docRef.get()
+            .addOnSuccessListener { document ->
+                if (document != null) {
+                    Log.d("document", "DocumentSnapshot data: ${document.data}")
+                    userName = document.getString("username").toString()
+                } else {
+                    Log.d("no document", "No such document")
+                }
+            }
+            .addOnFailureListener { exception ->
+                Log.d("error", "get failed with ", exception)
+            }
+
+        return userName
     }
 
+    /**
+     * Provide the user points from the firebase database
+     * @return userPoints
+     */
+    fun getUserPoints(): String{
+        val docRef = firestore.collection("users").document("testuser")
+        var userPoints: String = ""
+
+        docRef.get()
+            .addOnSuccessListener { document ->
+                if (document != null) {
+                    Log.d("document", "DocumentSnapshot data: ${document.data}")
+                    userPoints = document.getString("points").toString()
+                } else {
+                    Log.d("no document", "No such document")
+                }
+            }
+            .addOnFailureListener { exception ->
+                Log.d("error", "get failed with ", exception)
+            }
+
+        return userPoints
+    }
+
+    /**
+     * gets the list items from firebase
+     */
     private fun waitForListUpdate() {
         firestore.collection("lists/Top Fifteen Movies/MyListItems").addSnapshotListener{
-            snapshot, e ->
+                snapshot, e ->
             if(e != null){
                 Log.w(TAG, "Listen Failed", e)
                 return@addSnapshotListener
             }
+
             if(snapshot != null){
-                val allListItems = ArrayList<ListItem>()
-                val documents = snapshot.documents
-                documents.forEach{
+                if(snapshot.getDocumentChanges().size > 1 ){
```

Comment:
This if statement I added is significant. 
Since when a user votes firebase gets an update, this code will run.
The problem with that is that the code was reassigning the incoming data to mvm, which was causing the recycler view to re-render its list. 
This of course is not a good use of resources (and can also cause problems when trying to disable a vote button when a user clicks).
The cool thing is that when a user clicks, each click is sent to the database, and therefore there will only be one chance in the data coming in from firebase.
This line will filter out these cases and will avoid unnecessary re-rendering.